gui/src/app/pages/HomePage/HomePage.tsx

```diff
@@ -124,30 +124,47 @@ const LeftView: FunctionComponent<LeftViewProps> = () => {
   );
 };
 
-const DataEditor: FunctionComponent<{}> = () => {
+const DataEditor: FunctionComponent<unknown> = () => {
   const { data, update } = useContext(ProjectContext);
+  const defaultSourceDesc = "";
 
   const dataIsEdited = useMemo(() => {
     return data.dataFileContent !== data.ephemera.dataFileContent;
   }, [data.dataFileContent, data.ephemera.dataFileContent]);
 
+  const sourceDesc = useMemo(() => {
+    if (dataIsEdited) return defaultSourceDesc;
+    switch (data.meta.dataSource) {
+      case DataSource.GENERATED_BY_PYTHON: {
+        return "data.py";
+      }
+      case DataSource.GENERATED_BY_R: {
+        return "data.R";
+      }
+      case DataSource.GENERATED_BY_STALE_PYTHON: {
+        return "a prior version of data.py.";
+      }
+      case DataSource.GENERATED_BY_STALE_R: {
+        return "a prior version of data.R.";
+      }
+      default:
+        return defaultSourceDesc;
+    }
+  }, [dataIsEdited, data.meta.dataSource]);
+
   const dataMessage: ToolbarItem[] = useMemo(() => {
-    if (data.meta.dataSource === undefined || dataIsEdited) {
+    if (sourceDesc === defaultSourceDesc) {
       return [];
     } else {
       return [
         {
           type: "text",
-          label:
-            "Data is generated by data." +
-            (data.meta.dataSource === DataSource.GENERATED_BY_PYTHON
-              ? "py"
-              : "R"),
+          label: `Data is generated by ${sourceDesc}`,
```

Comment:
Thoughts on also changing the color when stale? To something more warning-ish?

Reply:
I could get behind that! Just didn't want to be too ambitious :smile: